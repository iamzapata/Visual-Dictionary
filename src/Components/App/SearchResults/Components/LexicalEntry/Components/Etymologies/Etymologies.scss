.etymologiesmargin: 20px 0;
