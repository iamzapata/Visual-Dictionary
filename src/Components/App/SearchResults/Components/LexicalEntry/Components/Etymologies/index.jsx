import React from "react"
import { arrayOf, string } from "prop-types"
import "./Etymologies.sass"

const Etymologies = ({ entry }) => {
  const etymologies = entry.entries.map(en => en.etymologies).filter(en => en)
  return (
    etymologies.length > 0 && (
      <div className="Etymologies">
        <h6 className="title is-6">Origin:</h6>
        <ul>{etymologies.map(et => <li key={et}>{et}</li>)}</ul>
      </div>
    )
  )
}

Etymologies.propTypes = {
  etymologies: arrayOf(string)
}

export default Etymologies
