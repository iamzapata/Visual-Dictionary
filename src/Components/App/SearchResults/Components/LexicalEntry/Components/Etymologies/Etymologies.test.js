import React from "react"
import { render, cleanup } from "react-testing-library"
import Etymologies from "./index"

describe("<Etymologies />", () => {
  afterEach(cleanup)

  const entry = {
    entries: [{ etymologies: ["a", "b"] }]
  }

  it("Should render self", () => {
    const { container } = render(<Etymologies entry={entry} />)
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
