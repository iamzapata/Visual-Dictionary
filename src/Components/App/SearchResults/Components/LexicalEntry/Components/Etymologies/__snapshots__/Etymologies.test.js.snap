// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Etymologies /> Should render self 1`] = `
<div
  class="Etymologies"
>
  <h6
    class="title is-6"
  >
    Origin:
  </h6>
  <ul>
    <li>
      a
      b
    </li>
  </ul>
</div>
`;
